src/scores/categorical/multicategorical_impl.py

```diff
@@ -187,10 +193,16 @@ def _single_category_score(
     # pylint: disable=unbalanced-tuple-unpacking
     fcst, obs = xr.align(fcst, obs)
 
-    # False Alarms
-    condition1 = (obs <= categorical_threshold) & (categorical_threshold < fcst)
-    # Misses
-    condition2 = (fcst <= categorical_threshold) & (categorical_threshold < obs)
+    if threshold_alignment == "right":
+        # False Alarms
+        condition1 = (obs <= categorical_threshold) & (categorical_threshold < fcst)
+        # Misses
+        condition2 = (fcst <= categorical_threshold) & (categorical_threshold < obs)
+    else:
+        # False Alarms
+        condition1 = (obs < categorical_threshold) & (categorical_threshold <= fcst)
+        # Misses
+        condition2 = (fcst < categorical_threshold) & (categorical_threshold <= obs)
```

Comment:
These look the correct way around to me, matching the description you have given for left and right alignment.